multi/imxrt-multi/helpers.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT helper macros
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Daniel Sawka
+ *
+ * %LICENSE%
+ */
+
+#ifndef _HELPERS_H_
+#define _HELPERS_H_
+
+#define CONCAT2(x, y)    x##y
+#define CONCAT3(x, y, z) x##y##z
+
+#define PIN2MUX(x) CONCAT2(pctl_mux_gpio_, x)
+#define PIN2PAD(x) CONCAT2(pctl_pad_gpio_, x)
+
+/* For the following macros, anything that expands to a numerical expression (or to an empty value) is fine.
+String literals don't work.
+
+/* True if X is defined, is not empty, and has a non-zero value. */
+#define ISTRUTHY(X) ((X + 0) != 0)
```

Comment:
Do you mean `#define ISTRUTHY(X) (defined(X) && (X != 0))`? This is undefined behavior (see `-Wexpansion-to-defined`) and also causes a syntax error if `X` is defined but empty. That's why `defined()` are used directly in `#if`s.

Reply:
thanks for bringing it to my attention, I wasn't aware of that. Seems the implementation for function-like macros in all major compilers is consistent (https://reviews.llvm.org/D15866) but it's still undefined behavior so let's not change it.

Maybe explicit comment regarding UB should be added here to avoid undesired changes in the future.

Reply:
still, might be good to add comment regarding WHY we can't add `defined(X)` here (eg. "using `defined(X)` in macro body is UB (see -Wexpansion-to-defined for details))

Reply:
Added a comprehensive (I think) comment